deque.go

```diff
@@ -432,3 +435,50 @@ func (q *Deque[T]) resize(newSize int) {
 	q.tail = q.count
 	q.buf = newBuf
 }
+
+// TakeSlice returns the contents of the queue as a contiguous slice,
+// leaving the queue empty.
+func (q *Deque[T]) TakeSlice() []T {
+	if q == nil {
```

Comment:
```suggestion
	if q == nil || q.count == 0{
```
This way an empty Deque does not need to give up its buffer and can use it for later.